.github/workflows/benchmark_execution.yml

```
@@ -91,10 +93,13 @@ jobs:
       - machine-type=${{ matrix.benchmark.device_name }}
     env:
       DEVICE_NAME: ${{ matrix.benchmark.device_name }}
+      SHARD_INDEX: ${{ matrix.benchmark.shard.index }}
+      SHARD_COUNT: ${{ matrix.benchmark.shard.count }}
       PLATFORM_ARCH: ${{ matrix.benchmark.host_environment.platform }}-${{ matrix.benchmark.host_environment.architecture }}
       E2E_TEST_ARTIFACTS_GCS_ARTIFACT_DIR: ${{ inputs.e2e-test-artifacts-gcs-artifact-dir }}
       E2E_TEST_ARTIFACTS_DIR: ${{ inputs.e2e-test-artifacts-dir }}
       BENCHMARK_RESULTS_DIR: benchmark-results
+      SHARDING_SUFFIX: ${{ matrix.benchmark.shard.count > 1 && format('-shard{0}of{1}', matrix.benchmark.shard.index, matrix.benchmark.shard.count) || '' }}
```

I believe we can write multiline expression, something like:
```suggestion
      SHARDING_SUFFIX: |
        ${{ matrix.benchmark.shard.count > 1
        && format('-shard{0}of{1}', matrix.benchmark.shard.index, matrix.benchmark.shard.count)
        || '' }}
```

---

Applied. I had to use `SHARDING_SUFFIX: |-` though because without the extra minus sign YAML adds a line break at the end which we don't want to have in the variable.